src/Handler/StreamHandler.php

```diff
@@ -210,7 +210,7 @@ private function drain(
         Psr7\copy_to_stream(
             $source,
             $sink,
-            strlen($contentLength) > 0 ? (int) $contentLength : -1
+            (strlen($contentLength) > 0 && (int) $contentLength > 0) ? (int) $contentLength : -1
```

Comment:
It would, but it's more error prone IMO, because you would still need to check if it's a numeric value.